controller/controller_manager.go

```diff
@@ -174,15 +175,39 @@ func GetGuaranteedResourceRequirement(ds *datastore.DataStore) (*corev1.Resource
 	}, nil
 }
 
-func isControllerResponsibleFor(controllerID string, ds *datastore.DataStore, name, preferredOwnerID, currentOwnerID string) bool {
+func isControllerResponsibleFor(controllerID string, ds *datastore.DataStore, name, preferredOwnerID, currentOwnerID string, getKubeNode bool) bool {
 	var err error
 	responsible := false
 
 	ownerDown := false
 	if currentOwnerID != "" {
-		ownerDown, err = ds.IsNodeDownOrDeleted(currentOwnerID)
-		if err != nil {
-			logrus.Warnf("Error while checking if object %v owner is down or deleted: %v", name, err)
+		if getKubeNode {
+			kubeNode, err := ds.GetKubernetesNode(currentOwnerID)
+			if err != nil {
+				// Kubernetes node has been removed from cluster.
+				if apierrors.IsNotFound(err) {
+					logrus.Warnf("Kubernetes node %v is not found", name)
+					ownerDown = true
+				} else {
+					// For other error don't trade as node down.
+					// Be consistent with ds.IsNodeDownOrDeleted.
+					logrus.Warnf("Kubernetes get node %v failed: %v, don't mark as node down", name, err)
+				}
+			} else {
+				kubeConditions := kubeNode.Status.Conditions
+				for _, con := range kubeConditions {
+					if con.Type == corev1.NodeReady {
```

Comment:
Node not ready can be caused by various reasons, for example, manager pod is down (https://github.com/longhorn/longhorn-manager/blob/84228670e96accb94a1597a69a0d7443b956f5ea/controller/node_controller.go#L332 ) or node has various reasons or pressure (https://github.com/longhorn/longhorn-manager/blob/84228670e96accb94a1597a69a0d7443b956f5ea/controller/node_controller.go#L385 ). 

Also, we shouldn't need `getKubeNode bool`.

This change can have more implications on how to decide which condition we should set for the node. For example, should we move the condition check to another function? We definitely won't want to duplicate the code here.